src/parser.js

```diff
@@ -265,16 +265,19 @@ export default class Parser {
   // a function (e.g. to create an element).
   withPath(...args) {
     let i;
+    const originalPath = _.clone(this.path);
```

Comment:
Agree, but there is more parts which should be refactorized.
especialy `parser.handleSwaggerExampleRequest()`is good adept.